code/modules/randomMaps/vaults.dm

```diff
@@ -102,6 +102,209 @@
 
 	message_admins("<span class='info'>Loaded space hobo shack [result ? "" : "un"]successfully.</span>")
 
+/proc/create_hell()
```

Comment:
it seems insane to create hell every shift when we rarely even use centcomm jail

just make this a fun verb of its own

Reply:
The difference there is that the centcomm jail is basically a no interaction totally bland cuck cube. Hell is on fire, has colour, you can scream your lungs out and, if he eventually implements it, the spirits and tortured bodies of shitters of olde.

Reply:
> The difference there is that the centcomm jail is basically a no interaction totally bland cuck cube. Hell is on fire, has colour, you can scream your lungs out and, if he eventually implements it, the spirits and tortured bodies of shitters of olde.

That doesn't mean it needs to be created automatically every shift.

Reply:
>That doesn't mean it needs to be created automatically every shift.

True. Perhaps the centcomm jail should be replaced with hell on each map?

Reply:
It's already coded to generate hell if required when the first person is sent to it. The automatic generation was probably left in by mistake.

Reply:
nah not a mistake, it freezes the game for 4 seconds at the lowest (was previously way longer until i did the fast turf init thing) so less people notice it at roundstart load

Reply:
Then perhaps you should remove the text from the OP stating that it's only generated when a person is first sent there. Also, pregenerating a whole Z-level to be used once a year or so is pretty unacceptable.

Reply:
The concept is great, but adding +4 seconds to load time every shift forever when it won't seriously be used is not good. If you really want to, you can map something onto z-2 manually where there is plenty of space.

Reply:
>you can map something onto z-2 manually where there is plenty of space

Careful to not map over the vault areas, some of the larger vaults, such as sokoban or the mothership lab take a LOT of space, you don't want the mapped hell to be overwritten by the vaults (or viceversa).

Reply:
Or you could generate it in the Z2 vault area
I don't remember creating a new Z-level taking four seconds, though. Maybe I'm just remembering incorrectly, but I swear it was much faster. Though it is highly hardware-dependent too, of course.

Reply:
If generating a new Z level every round is gonna be such a probblem, you could just as easily make Hell a corner of the Centcomm Z-Level or something
Smaller scale hell, same suffering
Ideally we'd be able to make it loop, just a tiny looping box of hell in the corner

Reply:
making it a dungeon to resolve performance issues